libs/pageserver_api/src/models.rs

```diff
@@ -227,6 +230,39 @@ pub struct TimelineInfo {
     pub state: TimelineState,
 }
 
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Serialize, Deserialize)]
+pub struct LayerMapInfo {
+    pub in_memory_layers: BTreeSet<InMemoryLayerInfo>,
+    pub historic_layers: BTreeSet<HistoricLayerInfo>,
+}
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Copy, Clone, Serialize, Deserialize)]
+#[serde(tag = "kind")]
+pub enum InMemoryLayerInfo {
+    Open { lsn_start: Lsn },
+    Frozen { lsn_start: Lsn, lsn_end: Lsn },
+}
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Serialize, Deserialize)]
+#[serde(tag = "kind")]
+pub enum HistoricLayerInfo {
+    Delta {
+        layer_file_name: String,
+        key_start: String,
```

Comment:
Why is this a `String`? We have a `impl fmt::Display for Key`

Reply:
I did not want to drag the whole `Key` with impls into `models` and change imports everywhere.

But sure, moved and fixed all places.

Reply:
If I add `#[serde_as(as = "DisplayFromStr")]` on top of the `key_start: Key,` and similar fields, I still don't get the `Display`-based serialization, interesting.

Reply:
TIL, `#[serde_as]` has to be the topmost annotation placed (or, at least above `#[derive(Serialize,Deserialize)]`, otherwise silently won't (de)serialize in the proper format.

Fixed that, need to add tests.